cmd/packaging/linux-appimage.go

```diff
@@ -40,45 +23,9 @@ func InitLinuxAppImage() {
 		"mv squashfs-root appimagetool && \\",
 		"rm appimagetool-x86_64.AppImage",
 		"ENV PATH=/opt/appimagetool/usr/bin:$PATH",
-	})
-
-	printInitFinished(packagingFormat)
-}
-
-func BuildLinuxAppImage(buildVersion string) {
-	projectName := pubspec.GetPubSpec().Name
-	packagingFormat := "linux-appimage"
-	tmpPath := getTemporaryBuildDirectory(projectName, packagingFormat)
-	defer func() {
-		err := os.RemoveAll(tmpPath)
-		if err != nil {
-			log.Errorf("Could not remove temporary build directory: %v", err)
-			os.Exit(1)
-		}
-	}()
-	log.Infof("Packaging AppImage in %s", tmpPath)
-
-	err := copy.Copy(build.OutputDirectoryPath("linux"), filepath.Join(tmpPath, "build"))
-	if err != nil {
-		log.Errorf("Could not copy build folder: %v", err)
-		os.Exit(1)
-	}
-	err = copy.Copy(packagingFormatPath(packagingFormat), filepath.Join(tmpPath))
-	if err != nil {
-		log.Errorf("Could not copy packaging configuration folder: %v", err)
-		os.Exit(1)
-	}
-
-	runDockerPackaging(tmpPath, packagingFormat, []string{"appimagetool", "."})
-
-	resultFileName := projectName + "-x86_64.AppImage"
-	outputFileName := projectName + "-" + buildVersion + ".AppImage"
-	outputFilePath := filepath.Join(build.OutputDirectoryPath("linux-appimage"), outputFileName)
-	err = copy.Copy(filepath.Join(tmpPath, resultFileName), outputFilePath)
-	if err != nil {
-		log.Errorf("Could not move AppImage file: %v", err)
-		os.Exit(1)
-	}
-
-	printPackagingFinished(packagingFormat)
+	},
+	buildOutputDirectory: "build",
+	packagingCommand:     "appimagetool .",
+	resultFileName:       "{{.projectName}}-x86_64.AppImage",
+	outputFileExtension:  "AppImage",
```

Comment:
Since `packagingCommand` is now always a templated script, we could maybe simplify things by removing resultFileName and always expect the script to output a file named `{{.projectName}}-{{.version}}.{{.outputFileExtension}}` ? Most scripts already output a file named like that.
I've seen `appimagetool` accepts a `[DESTINATION]` argument, but I don't know if it always appends `-x86_64.AppImage` to it? The other solution would be to `mv {{.projectName}}-x86_64.AppImage {{.projectName}}-{{.version}}.AppImage`